.github/workflows/docs-preview.yml

```diff
@@ -0,0 +1,150 @@
+name: Deploy Docs Preview
+
+on:
+  issue_comment:
+    types:
+      - created
+
+jobs:
+  check_auth:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.issue.pull_request && github.event.comment.body == '/docs' }}
+    steps:
+      - name: Check auth
+        id: check_auth
+        uses: actions/github-script@v6
+        with:
+          result-encoding: string
+          github-token: ${{ secrets.GH_TOKEN }}
+          script: |
+            const { owner, repo } = context.issue;
+            const { login: username } = context.payload.comment.user;
+            await github.rest.repos.checkCollaborator({
+              owner,
+              repo,
+              username,
+            });
+            github.rest.reactions.createForIssueComment({
+              owner,
+              repo,
+              comment_id: context.payload.comment.id,
+              content: "eyes",
+            });
+            const { number: pull_number } = context.payload.issue;
+            return pull_number;
+      - name: Not a collaborator
+        if: ${{ failure() }}
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GH_TOKEN }}
+          script: |
+            const { owner, repo, number: issue_number } = context.issue;
+            github.rest.reactions.createForIssueComment({
+              owner,
+              repo,
+              comment_id: context.payload.comment.id,
+              content: "-1",
+            });
+            github.rest.issues.createComment({
+              owner,
+              repo,
+              issue_number,
+              body: 'Docs previews can only be requested by collaborators. Please tag a maintainer to request a docs preview. ❌',
+            });
+    outputs:
+      pr_number: ${{ steps.check_auth.outputs.result }}
+
+  deploy-docs-preview:
+    runs-on: ubuntu-latest
+    needs: check_auth
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Bazel
+        run: |
+          wget -q https://github.com/bazelbuild/bazelisk/releases/download/v1.11.0/bazelisk-linux-amd64 -O bazelisk
+          chmod +x bazelisk
+          echo "$(pwd)" >> $GITHUB_PATH
+
+      - name: Build and Deploy Docs Preview
+        id: deploy
+        run: |
+          PR_NUMBER=${{ needs.check_auth.outputs.pr_number }}
+          echo "Deploying docs preview for PR #$PR_NUMBER"
+          STABLE_DOCS_BASE_PATH="pr/$PR_NUMBER" \
+          STABLE_ALGOLIA_SEARCH_API_KEY=${{ secrets.ALGOLIA_NEXT_SEARCH_API_KEY }} \
+          STABLE_ALGOLIA_SEARCH_APPID="D477I7TDXB" \
+          STABLE_ALGOLIA_SEARCH_INDEX="crawler_Player (Next)" \
+          bazel run --config=release //docs:gh_deploy -- --dest_dir "pr/$PR_NUMBER"
```

Comment:
What's the rationale for splitting up docs publishing for canaries vs main releases? We now have this logic in multiple places, across 2 CI systems. It's not a ton of overhead, but now we need to maintain secrets in two places as well. Couldn't we bake PR docs publishing into the release script that releases docs for main releases? Or should we move docs publishing out of that script and use GH actions entirely for it?

Reply:
I would also agree with Jeremiah. Can we just make this part of the Circle build as a parallel stage to publish? As it stands currently this would have to rerun the whole core/react compilation stage in this Action context